### ISSUE TITLE
[studio][studio-ui] Switch to use the API 2 get/write studio config.

### ISSUE THREAD

**Issue by yacdaniel** [craftercms/craftercms#4738]
# Feature Request

#### Is your feature request related to a problem? Please describe.
API's 1 are old and new mechanics were created in API 2. 

#### Describe the solution you'd like
API 1 `/api/1/services/api/1/site/get-configuration.json` and `/api/1/services/api/1/site/write-configuration.json` should be marked as deprecated and Studio should use the configuration API 2 `studio/api/2/configuration/get_configuration` and `studio/api/2/configuration/write_configuration`

#### Tasks
- [x] Deprecate old APIs
- [x] Switch UI to new APIs
- [x] Update Docs

#### Describe alternatives you've considered
N/A

#### Additional context
This applies for `v4.0.0`

**Comment by dejan-brkic** [comment]
Done
PR:
https://github.com/craftercms/studio/pull/2579

**Comment by alhambrav** [comment]
PR https://github.com/craftercms/docs/pull/1261 and  https://github.com/craftercms/docs/pull/1262

**Comment by dejan-brkic** [comment]
3.1.x
PR:
https://github.com/craftercms/studio/pull/2581

**Comment by yacdaniel** [comment]
@alhambrav Studio code changes were reverted in 3.1. Please rollback the 3.1 docs changes too.

**Comment by alhambrav** [comment]
PR https://github.com/craftercms/docs/pull/1291

**Comment by yacdaniel** [comment]
Verified fixed.

**Comment by yacdaniel** [comment]
@dejan-brkic content monitoring is not running because is looking for api 1 `getConfiguration` 
```
[INFO] 2021-07-06T12:00:01,289 [pool-11-thread-10] [core.JobRunShell] | Job DEFAULT.craftercms:/scripts/jobs/every-day/content-monitoring.groovy threw a JobExecutionException:  
org.quartz.JobExecutionException: Error executing script job at /scripts/jobs/every-day/content-monitoring.groovy
	at org.craftercms.engine.scripting.impl.ScriptJob.execute(ScriptJob.java:63) ~[classes/:4.0.0-SNAPSHOT]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202) [quartz-2.3.2.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: org.craftercms.engine.exception.ScriptException: No signature of method: org.craftercms.studio.impl.v1.service.site.SiteServiceImpl$$EnhancerBySpringCGLIB$$51a6a76.getConfiguration() is applicable for argument types: (String, String, Boolean) values: [erd2, site-config.xml, false]
	at org.craftercms.engine.scripting.impl.GroovyScript.execute(GroovyScript.java:74) ~[classes/:4.0.0-SNAPSHOT]
	at org.craftercms.engine.scripting.impl.ScriptJob.execute(ScriptJob.java:61) ~[classes/:4.0.0-SNAPSHOT]
	... 4 more
Caused by: groovy.lang.MissingMethodException: No signature of method: org.craftercms.studio.impl.v1.service.site.SiteServiceImpl$$EnhancerBySpringCGLIB$$51a6a76.getConfiguration() is applicable for argument types: (String, String, Boolean) values: [erd2, site-config.xml, false]
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:70) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:46) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:157) ~[groovy-3.0.7.jar:3.0.7]
	at scripts.api.impl.content.ContentMonitoring.doContentMonitoringForSite(ContentMonitoring.groovy:60) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107) ~[groovy-3.0.7.jar:3.0.7]
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:351) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:61) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203) ~[groovy-3.0.7.jar:3.0.7]
	at scripts.api.impl.content.ContentMonitoring$_doMonitoringForAllSites_closure1.doCall(ContentMonitoring.groovy:37) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107) ~[groovy-3.0.7.jar:3.0.7]
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263) ~[groovy-3.0.7.jar:3.0.7]
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035) ~[groovy-3.0.7.jar:3.0.7]
	at groovy.lang.Closure.call(Closure.java:412) ~[groovy-3.0.7.jar:3.0.7]
	at groovy.lang.Closure.call(Closure.java:428) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2318) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2303) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2356) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.dgm$202.invoke(Unknown Source) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:247) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139) ~[groovy-3.0.7.jar:3.0.7]
	at scripts.api.impl.content.ContentMonitoring.doMonitoringForAllSites(ContentMonitoring.groovy:34) ~[?:?]
	at scripts.api.impl.content.ContentMonitoring$doMonitoringForAllSites.call(Unknown Source) ~[?:?]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125) ~[groovy-3.0.7.jar:3.0.7]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148) ~[groovy-3.0.7.jar:3.0.7]
	at content-monitoring.run(content-monitoring.groovy:19) ~[?:?]
	at org.craftercms.engine.scripting.impl.GroovyScript.execute(GroovyScript.java:72) ~[classes/:4.0.0-SNAPSHOT]
	at org.craftercms.engine.scripting.impl.ScriptJob.execute(ScriptJob.java:61) ~[classes/:4.0.0-SNAPSHOT]
	... 4 more
```

**Comment by dejan-brkic** [comment]
Done
PR:

**Comment by yacdaniel** [comment]
Verified fixed in `4.0.0-SNAPSHOT`
